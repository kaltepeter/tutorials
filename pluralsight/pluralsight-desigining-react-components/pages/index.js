const indexPage = () => {
    return (<div>Hello from pluralsight</div>);
}

export default indexPage;