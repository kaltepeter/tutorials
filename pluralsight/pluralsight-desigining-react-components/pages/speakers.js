import react from "react";
import SpeakersRenderProps from "../src/components/SpeakerRenderProps";

const Speakers = () => {
    return (
        <SpeakersRenderProps>
            {({speakers})=> {
                return (
                    <div>
                        {speakers.map(({ imageSrc, name }) => {
                            return <img src={`images/${imageSrc}.jpg`} alt={name} key={imageSrc}></img>
                        })}
                    </div>
                );
            }}
        </SpeakersRenderProps>
    );
}

export default Speakers;