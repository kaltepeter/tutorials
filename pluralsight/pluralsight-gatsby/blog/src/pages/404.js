import React from "react"
import styles from "./404.module.scss"

export default () => {
  return (
    <div>
      <h1 className={styles.header}>Page not found</h1>
      <p className={styles.errorMessage}>
        The page you are looking for does not exist.
      </p>
    </div>
  )
}
