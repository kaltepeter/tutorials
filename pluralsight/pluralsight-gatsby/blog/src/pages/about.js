import React from "react"
import { Link } from "gatsby";

export default () => {
  return <div>
    <h1>This is the about page</h1>
    <div>
      <Link to="/">Home</Link> | <Link to="/about">About me</Link>
    </div>
    <p>Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.</p>
  </div>
}
