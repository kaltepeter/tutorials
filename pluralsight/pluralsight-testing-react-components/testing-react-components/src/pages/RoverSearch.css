.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 20vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
  background-image: url(./NASA_Mars_Rover.jpg);
  margin-bottom: 40px;
  background-size: cover;
background-position-y: center;
}

.App-header > h2 {
  background-color: black;
  padding: 0.2em;
}

.App-link {
  color: #61dafb;
}

.App-form .row {
  margin-top: 20px;
}

.App-img-result {
  margin-bottom: 20px;
  width: 100%;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
