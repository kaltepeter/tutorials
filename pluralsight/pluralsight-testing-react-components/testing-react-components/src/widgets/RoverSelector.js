import React from 'react';
import './roverselector.css';

function RoverSelector({ rovers, roversActive, roverSelection, onRoverSelection }) {
    const onClick = (roverName) => {
        onRoverSelection({ ...roverSelection, [roverName]: !roverSelection[roverName] });
    };

    return (
        <div className="row" style={{ textAlign: "center" }}>
        {rovers.map((rover) => <div data-testid={`rover-div-${rover.name.toLowerCase()}`} className={`col-4 ${roversActive[rover.name.toLowerCase()] ? '' : 'RoverSelector-inactive'}`} key={rover.name}>
            <strong>{rover.name}</strong><br/>
            <small>{rover.activePeriod[0]} - {rover.activePeriod[1]}</small>
            <p>
                <input type="checkbox" data-testid="rover-selected" checked={roverSelection[rover.name.toLowerCase()]} onChange={() => { onClick(rover.name.toLowerCase()) }}></input>
            </p>
        </div>)}
        </div>
    );
}

export default RoverSelector;