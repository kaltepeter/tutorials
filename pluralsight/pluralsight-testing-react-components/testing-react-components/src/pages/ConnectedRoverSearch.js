import { connect } from 'react-redux';
import debounce from 'lodash/debounce';
import * as nasa from '../io/nasa';
import RoverSearch from './RoverSearch';

export const rovers = [
  { name: "Spirit", activePeriod: ["2004-01-05", "2010-03-21"] },
  { name: "Opportunity", activePeriod: ["2004-01-26", "2018-06-11"] },
  { name: "Curiosity", activePeriod: ["2012-08-07", "2019-09-28"] }
];

export const cameras = {
  FHAZ: "Front Hazard Avoidance Camera",
  RHAZ: "Rear Hazard Avoidance Camera",
  MAST: "Mast Camera",
  CHEMCAM: "Chemistry and Camera Complex",
  MAHLI: "Mars Hand Lens Imager",
  MARDI: "Mars Descent Imager",
  NAVCAM: "Navigation Camera",
  PANCAM: "Panoramic Camera",
  MINITES: "Miniature Thermal Emission Spectrometer (Mini-TES)"
};

export const roverSelection = {
  spirit: true,
  opportunity: true,
  curiosity: true
};

export const stateChunk = {
  rovers: rovers, 
  cameras: cameras, 
  roverSelection,
  earth_date: "2019-09-28",
  camera: "FHAZ",
  photos: [{"id":689597,"sol":2540,"camera":{"id":20,"name":"FHAZ","rover_id":5,"full_name":"Front Hazard Avoidance Camera"},"img_src":"https://mars.nasa.gov/msl-raw-images/proj/msl/redops/ods/surface/sol/02540/opgs/edr/fcam/FLB_622990123EDR_F0763002FHAZ00341M_.JPG","earth_date":"2019-09-28","rover":{"id":5,"name":"Curiosity","landing_date":"2012-08-06","launch_date":"2011-11-26","status":"active","max_sol":2540,"max_date":"2019-09-28","total_photos":366206,"cameras":[{"name":"FHAZ","full_name":"Front Hazard Avoidance Camera"},{"name":"NAVCAM","full_name":"Navigation Camera"},{"name":"MAST","full_name":"Mast Camera"},{"name":"CHEMCAM","full_name":"Chemistry and Camera Complex"},{"name":"MAHLI","full_name":"Mars Hand Lens Imager"},{"name":"MARDI","full_name":"Mars Descent Imager"},{"name":"RHAZ","full_name":"Rear Hazard Avoidance Camera"}]}},{"id":689598,"sol":2540,"camera":{"id":20,"name":"FHAZ","rover_id":5,"full_name":"Front Hazard Avoidance Camera"},"img_src":"https://mars.nasa.gov/msl-raw-images/proj/msl/redops/ods/surface/sol/02540/opgs/edr/fcam/FRB_622990123EDR_F0763002FHAZ00341M_.JPG","earth_date":"2019-09-28","rover":{"id":5,"name":"Curiosity","landing_date":"2012-08-06","launch_date":"2011-11-26","status":"active","max_sol":2540,"max_date":"2019-09-28","total_photos":366206,"cameras":[{"name":"FHAZ","full_name":"Front Hazard Avoidance Camera"},{"name":"NAVCAM","full_name":"Navigation Camera"},{"name":"MAST","full_name":"Mast Camera"},{"name":"CHEMCAM","full_name":"Chemistry and Camera Complex"},{"name":"MAHLI","full_name":"Mars Hand Lens Imager"},{"name":"MARDI","full_name":"Mars Descent Imager"},{"name":"RHAZ","full_name":"Rear Hazard Avoidance Camera"}]}}]
};

export const reducer = (state = stateChunk, action) => {
  switch (action.type) {
    case "CRITERIA": {
      return {
        ...state,
        ...action.criteria
      };
    }
    case "CRITERIA_CHANGE_FULFILLED": {
      return {
        ...state,
        photos: action.payload.map((response) => response.photos).flat()
      };
    }
    default: return state;
  }
};

const ConnectedRoverSearch = connect(
  (state) => state,
  (dispatch) => ({
    onCriteriaChange: (criteria) => {
      dispatch({type: "CRITERIA", criteria});
    },
    onSearch: debounce((criteria) => {
      dispatch({type: "CRITERIA_CHANGE", criteria, payload: nasa.searchImages(criteria) });
    }, 500)
  })
)(RoverSearch);

export default ConnectedRoverSearch;
