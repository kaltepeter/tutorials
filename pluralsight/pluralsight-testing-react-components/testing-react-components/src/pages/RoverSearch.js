import React from 'react';
import "./RoverSearch.css";
import DateSlider from '../widgets/DateSlider';
import RoverSelector from '../widgets/RoverSelector';
import CameraSelection from '../widgets/CameraSelection';
import {isActive} from '../services/sols';
  
function RoverSearch(props) {
    const {rovers, cameras, camera, roverSelection, earth_date, photos, onCriteriaChange, onSearch} = props;
    const setCriteriaAndPublish = (newCriteria) => {
        const merged = {...props, ...newCriteria};
        onCriteriaChange(merged);
        onSearch(merged);
    };
    const roversActive = rovers.reduce((acc, curr, i) => {
        acc[curr.name.toLowerCase()] = isActive(curr.activePeriod[0], curr.activePeriod[1], earth_date);
        return acc;
    }, {});

    return (
        <div>
        <header className="App-header">
            <h2>Mars Rover Images</h2>
        </header>
        <div className="container-fluid App-form">
            <div className="row">
            <form className="col-10 offset-1">
                <DateSlider earth_date={earth_date} onDateChanged={(date)=> {setCriteriaAndPublish({earth_date: date});}} />
                <RoverSelector roversActive={roversActive} rovers={rovers} roverSelection={roverSelection} onRoverSelection={(selected)=> {setCriteriaAndPublish({roverSelection: selected});}} />
                <CameraSelection camera={camera} cameras={cameras} onCameraSelected={(selectedCamera) => {setCriteriaAndPublish({ camera: selectedCamera });}} />
            </form>
            </div>
            <div className="row">
            <div className="col-10 offset-1">
            <div className="row">
                {photos.length === 0 
                ? <p>No results</p> 
                : photos.map((photo) => 
                    <div key={photo.id} className="col-lg-4 col-12">
                        <img data-testid="rover-image" className="App-img-result" src={photo.img_src} alt="Mars"/>
                    </div>
                    )
                }
                </div>
            </div>
            </div>
        </div>
        </div>
    );
}

export default RoverSearch;