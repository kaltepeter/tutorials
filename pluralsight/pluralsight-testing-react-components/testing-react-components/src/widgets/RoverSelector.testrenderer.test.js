import React from 'react';
import TestRenderer from 'react-test-renderer';
import RoverSelector from './RoverSelector';
import {rovers} from '../pages/ConnectedRoverSearch';

describe('RoverSelector', () => {
    describe('rendering', () => {
        describe('selection', () => {
            describe('all selected', () => {
                it('should select all rovers', () => {
                    const all = {spirit: true, opportunity: true, curiosity: true};
                    const tr = TestRenderer.create(<RoverSelector rovers={rovers} roversActive={all} roverSelection={all} onRoverSelection={() => {}} />);
                    const inputs = tr.root.findAllByProps({
                        "data-testid": 'rover-selected'
                    });
                    inputs.forEach((input) => {
                        expect(input.props.checked).toBe(true);
                    });
                });
            });
            describe('none selected', () => {
                const none = {spirit: false, opportunity: false, curiosity: false};
                const tr = TestRenderer.create(<RoverSelector rovers={rovers} roversActive={none} roverSelection={none} onRoverSelection={() => {}} />);
                const inputs = tr.root.findAllByProps({
                    "data-testid": 'rover-selected'
                });
                inputs.forEach((input) => {
                    expect(input.props.checked).toBe(false);
                });
            })
        });

        describe('activation', () => {
            const inactiveCssClassExpression = /.*RoverSelector-inactive/;
            describe('all active', () => {
                it('should not have the inactive class', () => {
                    const all = {spirit: true, opportunity: true, curiosity: true};
                    const tr = TestRenderer.create(<RoverSelector rovers={rovers} roversActive={all} roverSelection={all} onRoverSelection={() => {}} />);
                    const divs = tr.root.findAll((instance) => {
                        return (instance.props['data-testid'] || '').startsWith('rover-div-');
                    });
                    divs.forEach((div) => {
                        expect(div.props.className).not.toMatch(inactiveCssClassExpression);
                    });
                });
            });

            describe('all inactive', () => {
                it('should have the inactive class', () => {
                    const none = {spirit: false, opportunity: false, curiosity: false};
                    const tr = TestRenderer.create(<RoverSelector rovers={rovers} roversActive={none} roverSelection={none} onRoverSelection={() => {}} />);
                    const divs = tr.root.findAll((instance) => {
                        return (instance.props['data-testid'] || '').startsWith('rover-div-');
                    });
                    divs.forEach((div) => {
                        expect(div.props.className).toMatch(inactiveCssClassExpression);
                    });
                });
            });
        });
    });
});