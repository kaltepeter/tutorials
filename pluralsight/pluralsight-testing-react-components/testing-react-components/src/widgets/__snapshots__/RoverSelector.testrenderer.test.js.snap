// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoverSelector rendering should not change 1`] = `
<div
  className="row"
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <div
    className="col-4 "
    data-testid="rover-div-spirit"
  >
    <strong>
      Spirit
    </strong>
    <br />
    <small>
      2004-01-05
       - 
      2010-03-21
    </small>
    <p>
      <input
        checked={true}
        data-testid="rover-selected"
        onChange={[Function]}
        type="checkbox"
      />
    </p>
  </div>
  <div
    className="col-4 "
    data-testid="rover-div-opportunity"
  >
    <strong>
      Opportunity
    </strong>
    <br />
    <small>
      2004-01-26
       - 
      2018-06-11
    </small>
    <p>
      <input
        checked={true}
        data-testid="rover-selected"
        onChange={[Function]}
        type="checkbox"
      />
    </p>
  </div>
  <div
    className="col-4 "
    data-testid="rover-div-curiosity"
  >
    <strong>
      Curiosity
    </strong>
    <br />
    <small>
      2012-08-07
       - 
      2019-09-28
    </small>
    <p>
      <input
        checked={true}
        data-testid="rover-selected"
        onChange={[Function]}
        type="checkbox"
      />
    </p>
  </div>
</div>
`;
